### ISSUE TITLE
TxSigner.SignJson should throw an exception if json contains unknown field

### ISSUE THREAD

**Issue by matejr** [ripple/ripple-dot-net#20]
This code should throw an exception. 

```C#
	var signer = TxSigner.FromSecret("snR7czZRBW5tRPTNsnhS1UpY3vx5X");
	var account = "r99V1djgizxfcscPUpUbbR2C2akai7UXDe";
	
	var r2 = signer.SignJson(JObject.FromObject(new { 
                  TransactionType = "AccountSet", 
                  Account = account, 
                  Fee = "100", 
                  Sequence = 10, 

                  // unknown field
                  UnknownField = 123121 }));
```
Latest commits didn't fix this "UnknownField" field problem.

**Comment by sublimator** [comment]
It's intentional, sorry

**Comment by sublimator** [comment]
Will be working on it more tomorrow, as well as typed transactions.

Hold up :)

**Comment by sublimator** [comment]
I added a `strict` arg to StObject.FromJson and used that from inside TxSigner. Many times in the past it's been useful to be able to just ignore json fields, but I think you're correct in that it makes sense here :)
The TxFormat.Validate routine will only catch missing required fields, and if someone typos an optional field, that would not be fun.

I would like to hear your thoughts about typed transactions. I saw you used interfaces, and UnsignedX.
